### ISSUE TITLE
Loss of three bytes of data

### ISSUE THREAD

**Issue by Jeksenc** [tendai98/ESP8266-CAN#4]
Hello.
When using Python scripts, the data is shifted to the right with a loss of bytes. The first three bytes are filled with 00, and the last ones are not displayed. 
I connected another CAN sniffer in parallel and saw that the data was cut off by three bytes.
wrong:
"data": "000000**1d00010700**"
right:
"data": "**1d00010700**2a0bff" (2a0bff - lost)

**Comment by epe** [comment]
Maybe you need a Level Converter? ESP8266 uses 3,3V and afaik the MCP2515 needs a 5V signal voltage - that could explain that.

**Comment by tendai98** [comment]
> Hello. When using Python scripts, the data is shifted to the right with a loss of bytes. The first three bytes are filled with 00, and the last ones are not displayed. I connected another CAN sniffer in parallel and saw that the data was cut off by three bytes. wrong: "data": "000000**1d00010700**" right: "data": "**1d00010700**2a0bff" (2a0bff - lost)

Let me investigate the issue to see if the bytes are being lost at the buffer in the MCU or something I am doing wrong in the Python script

**Comment by dreamer1048576** [comment]
Would you share your related code ?

**Comment by tendai98** [comment]
> Would you share your related code ?

I am debugging the hardware code , especially the code that is handling the frames: 
- https://github.com/tendai98/ESP8266-CAN/blob/main/ESP8266-CAN/can.h

I am also looking at the scanning script:
- https://github.com/tendai98/ESP8266-CAN/blob/main/utils/can_scan.py

**Comment by Jeksenc** [comment]
I used your code.
I saved the log. You can see that the data starts with zeros. I connected the Arduino Nano via USB and saw the full data.

![Image](https://github.com/user-attachments/assets/0e22ee9b-b373-4533-88d5-607fe6932ef3)
[log1.txt](https://github.com/user-attachments/files/24304224/log1.txt)

**Comment by tendai98** [comment]
> I used your code. I saved the log. You can see that the data starts with zeros. I connected the Arduino Nano via USB and saw the full data.
> 
> ![Image](https://github.com/user-attachments/assets/0e22ee9b-b373-4533-88d5-607fe6932ef3) [log1.txt](https://github.com/user-attachments/files/24304224/log1.txt)

Yeah I am noticing that we are missing 3 bytes from the data and I am startiung to assuming that it could that my pointers could be off by 3 interms of addressing because its  Python script issue

**Comment by tendai98** [comment]
> I used your code. I saved the log. You can see that the data starts with zeros. I connected the Arduino Nano via USB and saw the full data.
> 
> ![Image](https://github.com/user-attachments/assets/0e22ee9b-b373-4533-88d5-607fe6932ef3) [log1.txt](https://github.com/user-attachments/files/24304224/log1.txt)

Okay here is what I did, I changed the pointer offset to start 3 bytes ahead. Sadly I am on holiday so I dont have the hardware with me and cant run tests but let me know how it works if you manage to run it
